Week5CheckSubtree.cpp

```diff
@@ -0,0 +1,174 @@
+/*
+Given two non-empty binary trees s and t, check whether tree t has exactly the same structure and node values with a subtree of s.
+A subtree of s is a tree consists of a node in s and all of this node's descendants.
+
+SOLUTION:
+1. Create the trees s and t taking the level order traversal as the output.
+2. Check for the NULL cases
+	2.1 if s and t both are NULL
+		return true
+	2.2 if s is NULL
+		return false
+	2.3 if t is NULL
+		return true
+3. if s->val==t->val
+	Repeat the comparision for the rest of the s and t trees.
+4.  Check for subtree either on s->left or s->right
+
+*/
+#include <bits/stdc++.h>
+using namespace std;
+
+struct node
+{
+    int val;
+    struct node* left;
+    struct node* right;
+}*head_s,*head_t;
+
+struct node* newNode(int val)
+{
+    struct node* ptr = (struct node*)malloc(sizeof(node));
+    ptr->val = val;
+    ptr->left = ptr->right = NULL;
+    return ptr;
+}
+
+node* insertLevelOrder(int arr[], node* root,
+                       int i, int n)
+{
+    if (i < n)
+    {
+        node* temp = newNode(arr[i]);
+        root = temp; 
+        root->left = insertLevelOrder(arr,root->left, 2 * i + 1, n);
+        root->right = insertLevelOrder(arr,root->right, 2 * i + 2, n);
+    }
+    return root;
+}
+
+bool isSubtree(struct node* s, struct node* t) {
+    
+    if(s==NULL && t==NULL)
+        return true;
+    if(s==NULL || t==NULL)
+        return false;
+    
+    if(s->val==t->val&& isSubtree(s->left,t->left)&&isSubtree(s->right,t->right))
+        return true;
+            
+    return isSubtree(s->left,t) || isSubtree(s->right,t);
+       
+}
+
+bool checkSubtree(struct node* s, struct node* t)
+{
+	if (t == NULL) {
```

Comment:
Inconsistent spacing here, like the if is indented too much or in line 75, the int is indented too much.

Love the test names though, great improvement!

Reply:
Thanks @omusavi for the review.